scripts/migrate-users.py

```diff
@@ -0,0 +1,209 @@
+#!./venv/bin/python
+"""
+This script migrates users from the Django user models to the bundle service
+database.
+
+1. Load user data from the Django database
+2. Generate a new id for each user using the uuid scheme
+3. Update existing rows in the bundle service users table, using the Django
+   user models as a source of truth, leaving only the disk/time quota and usage
+   columns intact.
+4. Insert remaining users into the bundle service users table, filling in the
+   configured default quota information.
+5. Mark all users in the Django database as inactive, which means that they
+   will not be picked up in future runs of the migration script.
+
+By default, this script runs in dry-run mode, i.e. it prints verbose output
+such as the generated SQL queries but does not make changes to the databases.
+When you're ready to perform the migration, run with the '-f' flag.
+
+This scripts assumes that you are running it from the codalab-cli directory.
+"""
+
+import os
+import sys
+sys.path.append('.')
+
+from sqlalchemy import (
+    select,
+    bindparam,
+)
+
+from codalab.common import UsageError
+from codalab.model.tables import user as cl_user
+from codalab.lib.codalab_manager import (
+    CodaLabManager,
+    read_json_or_die,
+    print_block
+)
+
+
+class DryRunAbort(Exception):
+    """Raised at end of transaction of dry run."""
+    def __str__(self):
+        return """
+        This was a dry run, no migration occurred. To perform full migration,
+        run again with `-f':
+
+            %s -f
+        """.rstrip() % sys.argv[0]
+
+
+dry_run = False if len(sys.argv) > 1 and sys.argv[1] == '-f' else True
+
+manager = CodaLabManager()
+model = manager.model()
+CODALAB_HOME = manager.codalab_home
+
+# Turn on query logging
+model.engine.echo = True
+
+
+###############################################################
+# Configure connection to Django database
+###############################################################
+django_config = read_json_or_die(os.path.join(CODALAB_HOME,
+                                              'website-config.json'))
+
+# Use default settings as defined in codalab-worksheets
+if 'database' not in django_config:
+    django_config['database'] = {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': 'codalab.sqlite3',
+    }
+
+# Create database connections
+if django_config['database']['ENGINE'] == 'django.db.backends.mysql':
+    import MySQLdb
+    from MySQLdb.cursors import DictCursor
+    db_params = {
+        'db': django_config['database']['NAME'],
+        'user': django_config['database']['USER'],
+        'passwd': django_config['database']['PASSWORD'],
+        'cursorclass': DictCursor,
+    }
+    if django_config['database'].get('HOST', None):
+        db_params['host'] = django_config['database']['HOST']
+    if django_config['database'].get('PORT', None):
+        db_params['port'] = django_config['database']['PORT']
```

Comment:
Tried testing this on wstest. The deployment.config file is a bit weird. It specifies the port as a string.